actions/gha-workflow-validator/src/index.ts

```diff
@@ -0,0 +1,72 @@
+import * as core from "@actions/core";
+import * as github from "@actions/github";
+import { commentOnPrOrUpdateExisting, deleteCommentOnPRIfExists, getComparison } from "./github.js";
+import { validateActionReferenceChanges } from "./action-reference-validations.js";
+import { filterForGithubWorkflowChanges, formatGithubComment, parseAllAdditions } from "./utils.js";
+import { COMMENT_HEADER } from "./strings.js";
+
+(async () => {
+  const { token, owner, repo, base, head, prNumber } = getInvokeContext();
+  const octokit = github.getOctokit(token);
+
+  const allFiles = await getComparison(octokit, owner, repo, base, head);
+  const ghaWorkflowFiles = filterForGithubWorkflowChanges(allFiles);
+  const ghaWorkflowPatchAdditions = parseAllAdditions(ghaWorkflowFiles);
+
+  const containsWorkflowModifications = ghaWorkflowPatchAdditions.some(file => {
+    return (file.filename.startsWith('.github/workflows') || file.filename.startsWith('.github/actions')) && (file.filename.endsWith('.yml') || file.filename.endsWith('.yaml'))
+  });
+
+  if (containsWorkflowModifications) {
+    const actionReferenceValidations = await validateActionReferenceChanges(octokit, ghaWorkflowPatchAdditions)
+    const validationFailed = actionReferenceValidations.some(validation => validation.lineValidations.length > 0);
+    const invokedThroughPr = prNumber !== undefined;
+
+    if (validationFailed && invokedThroughPr) {
+      const errorMessage = formatGithubComment(actionReferenceValidations, owner, repo, head);
+      commentOnPrOrUpdateExisting(octokit, owner, repo, prNumber, errorMessage);
+      return core.setFailed("Errors found in workflow files. See comment on for details.");
+    } else if (validationFailed) {
+      // If the action is not invoked through a PR, we can't comment on the PR
+      return core.setFailed("Errors found in workflow files.");
+    } else if (!validationFailed && invokedThroughPr) {
+      deleteCommentOnPRIfExists(octokit, owner, repo, prNumber);
+    }
+  }
+})().catch((err) => {
+  core.error("Uncaught Error - " + err);
+  core.setFailed(err.message);
```

Comment:
Not sure that we need this, we can let the process crash. GHA should still set a failure and spit out a stacktrace

Reply:
Remvoed